### ISSUE TITLE
[Suggestion] Circular Padding

### ISSUE THREAD

**Issue by wolfrev0** [tuxedcat/openvino#1]
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html
pytorch has padding_mode=circular

https://docs.openvino.ai/latest/openvino_docs_ops_movement_Pad_1.html
https://docs.openvino.ai/latest/api/ngraph_python_api/_autosummary/ngraph.pad.html
OpenVINO doesn't have circular padding

Circular padding is useful when environment is circular.
https://github.com/tuxedcat/A2C
https://github.com/tuxedcat/PER-D3QN
In these cases, circular padding preserves border information better than zero padding.

**Comment by wolfrev0** [comment]
/home/felixkim/openvino/src/plugins/intel_gpu/src/plugin/ops/pad.cpp
/home/felixkim/openvino/src/plugins/intel_gpu/include/intel_gpu/primitives/border.hpp
/home/felixkim/openvino/src/plugins/intel_gpu/src/graph/border.cpp
/home/felixkim/openvino/src/plugins/intel_gpu/include/intel_gpu/plugin/program.hpp
/home/felixkim/openvino/src/plugins/intel_gpu/src/plugin/program.cpp
/home/felixkim/openvino/src/plugins/intel_gpu/src/graph/graph_optimizer/handle_input_padding.cpp

class T:public parent<T>{};// CRTP
